src/layout.cpp

```diff
@@ -765,6 +736,14 @@ static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XM
   return [fn,type,title,subtitle](LayoutParser &parser,const XMLHandlers::Attributes &attr) { (parser.*fn)(attr,type,title(),subtitle()); };
 }
 
+static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XMLHandlers::Attributes &,LayoutDocManager::LayoutPart,QCString,LayoutNavEntry::Kind),
+                                        LayoutDocManager::LayoutPart part,
+                                        QCString scope,
+                                        LayoutNavEntry::Kind nav
+                                       )
+{
+  return [fn,part,scope,nav](LayoutParser &parser,const XMLHandlers::Attributes &attr) { (parser.*fn)(attr,part,scope,nav); };
+}
```

Comment:
I cannot express how much I hate this source layout. I know, I know: It mirrors the other functions above it exactly, and code changes should always follow the prevailing style... I didn't say it was _wrong_, I said I hate it. I hate the way **all** of them are formatted.

Would a PR that reformatted them to something along these lines be considered for merging? Or is it not worth bothering, because the devs are fine with the current ultra-widescreen abomination? (After all, everyone's entitled to their opinion, however wrong it may be. :wink:)

```c++
static ElementCallbacks::StartCallback
startCb(
    void (LayoutParser::*fn)(
        const XMLHandlers::Attributes&,
        LayoutDocManager::LayoutPart,
        QCString,
        LayoutNavEntry::Kind),
    LayoutDocManager::LayoutPart part,
    QCString scope,
    LayoutNavEntry::Kind nav)
{
  return [fn,part,scope,nav](
      LayoutParser &parser,
      const XMLHandlers::Attributes &attr)
      {
          (parser.*fn)(attr,part,scope,nav);
      };
}
```

Or perhaps a compromise, like...

```c++
static ElementCallbacks::StartCallback
startCb(
  void (LayoutParser::*fn)(const XMLHandlers::Attributes&,LayoutDocManager::LayoutPart,QCString,LayoutNavEntry::Kind),
  LayoutDocManager::LayoutPart part,
  QCString scope,
  LayoutNavEntry::Kind nav)
{
  return [fn,part,scope,nav](LayoutParser &parser,const XMLHandlers::Attributes &attr) {
    (parser.*fn)(attr,part,scope,nav);
  };
}
```

Reply:
Consistency in the layout a a very good thing. The way how the layout of the code should look like is always a source for a good dispute between developers and all layouts have their pros and cons.
Also the length of a line is a problem, in the past it was restricted to 72 / 80 characters later 132 characters, there are only a few language standards that limit the line length.

Looking at:
```
static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XMLHandlers::Attributes &,LayoutDocManager::LayoutPart,QCString,LayoutNavEntry::Kind),
                                        LayoutDocManager::LayoutPart part,
                                        QCString scope,
                                        LayoutNavEntry::Kind nav
                                       )
```
I would even prefer:
```
static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XMLHandlers::Attributes &,LayoutDocManager::LayoutPart,QCString,LayoutNavEntry::Kind),
                                               LayoutDocManager::LayoutPart part,
                                               QCString scope,
                                               LayoutNavEntry::Kind nav
                                              )
```
or 
```
static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XMLHandlers::Attributes &,
                                                                        LayoutDocManager::LayoutPart,QCString,
                                                                        LayoutNavEntry::Kind
                                                                       ),
                                               LayoutDocManager::LayoutPart part,
                                               QCString scope,
                                               LayoutNavEntry::Kind nav
                                              )
```
or
```
static ElementCallbacks::StartCallback startCb(void (LayoutParser::*fn)(const XMLHandlers::Attributes &,
                                                                        LayoutDocManager::LayoutPart,QCString,
                                                                        LayoutNavEntry::Kind),
                                               LayoutDocManager::LayoutPart part,
                                               QCString scope,
                                               LayoutNavEntry::Kind nav)
```
and to be honest I definitely don't like a style like:
```
static ElementCallbacks::StartCallback
startCb(
```
where the `static` and return type are on a different line than the name of the function.